server/flight_plan.go

```diff
@@ -0,0 +1,160 @@
+	err := json.NewDecoder(r.Body).Decode(&body)
+	if err != nil {
+		http.Error(w, "malformed JSON", http.StatusBadRequest)
+		return
+	}
+	user := middleware.MustGetUserFromContext(r.Context())
+	params.Users = user.ID
+	params.FlightNumber = body.FlightNumber
+
+	flightPlan, err := a.Queries.CreateFlightPlan(context.Background(), params)
+	if err != nil {
+		log.Error().AnErr("CreateFlightPlan", err).Msg("Failed to create flight plan")
+		http.Error(w, "Failed to create flight plan", http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusCreated)
+	if err := json.NewEncoder(w).Encode(flightPlan); err != nil {
+		http.Error(w, "Failed to encode response", http.StatusInternalServerError)
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+}
+
+func (a *App) GetFlightPlansHandler(w http.ResponseWriter, r *http.Request) {
+	user := middleware.MustGetUserFromContext(r.Context())
+
+	flightPlans, err := a.Queries.GetFlightPlans(context.Background(), user.ID)
+	if err != nil {
+		log.Error().AnErr("GetFlightPlans", err).Msg("Failed to get flight plans")
+		http.Error(w, "Failed to get flight plans", http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(flightPlans); err != nil {
+		http.Error(w, "Failed to encode response", http.StatusInternalServerError)
+	}
+}
+
+func (a *App) GetFlightPlanHandler(w http.ResponseWriter, r *http.Request) {
+	user := middleware.MustGetUserFromContext(r.Context())
+	param, err := strconv.Atoi(chi.URLParam(r, "id"))
+	if err != nil {
+		http.Error(w, "Invalid Flight ID", http.StatusBadRequest)
+		return
+	}
+	flightPlanID := int32(param)
+	params := sqlc.GetFlightPlanParams{
+		Users: user.ID,
+		ID:    flightPlanID,
+	}
+
+	flightPlans, err := a.Queries.GetFlightPlan(context.Background(), params)
+	if err != nil {
+		log.Error().AnErr("GetFlightPlans", err).Msg("Failed to get flight plans")
+		http.Error(w, "Failed to get flight plans", http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(flightPlans); err != nil {
+		http.Error(w, "Failed to encode response", http.StatusInternalServerError)
+	}
+}
+
+func (a *App) DeleteFlightPlan(w http.ResponseWriter, r *http.Request) {
+	_ = middleware.MustGetUserFromContext(r.Context())
+	planID, err := strconv.Atoi(chi.URLParam(r, "id"))
+	if err != nil {
+		http.Error(w, "Invalid Flight Plan ID", http.StatusBadRequest)
+		return
+	}
+
+	flightPlan, err := a.Queries.DeleteFlightPlan(context.Background(), int32(planID))
+	if err != nil {
+		log.Error().AnErr("DeleteFlightPlan", err).Msg("Failed to delete flight plan")
+		http.Error(w, "Failed to delete flight plan", http.StatusInternalServerError)
+		return
+	}
+
+	if err := json.NewEncoder(w).Encode(flightPlan); err != nil {
+		http.Error(w, "Failed to encode response", http.StatusInternalServerError)
+	}
```

Comment:
Usually, for deletes, I use 204 No Content.

Returning a deleted entity doesn't make sense in most cases unless you're doing stuff with queues.